man/systemd.special.xml

```diff
@@ -667,6 +671,17 @@
           default).</para>
         </listitem>
       </varlistentry>
+      <varlistentry>
+        <term><filename>wakeup.target</filename></term>
+        <listitem>
+          <para>A special target unit that is started after
+          <filename>suspend.target</filename>,
+          <filename>hibernate.target</filename> and
+          <filename>hybrid-sleep.target</filename>.  This can be used
+          to pull in units and order them after the system wakes up
+          from sleep.</para>
+        </listitem>
```

Comment:
So there's a problem here. If a service is WantedBy=wakeup.target, Before=wakeup.target, it is not ordered relative to the systemd.sleep service in any way. Systemd is free to start it before the suspend happens. There are three solutions that come to mind:
1. tell users to specify also After=hibernate.target sleep.target hybrid-sleep.target
2. add a new target like wakeup-pre.target and order it After=hibernate.target sleep.target hybrid-sleep.target, WantedBy=wakup.target, and tell users to add After=wakeup-pre.target (this way they don't have to list all three targets)
3. revert the ordering of services wrt. to wakeup.target and tell users to specify WantedBy=wakeup.target, After=wakeup.target.

Actually I like 3. the best — it is concise and doesn't require the introduction of a new target.

It would be great if you could also add two examples to the man page: a pre-sleep service that hooks into sleep.target and a post-sleep service that hooks into wakeup.target. Otherwise I'm pretty sure people will screw this up.